core/modules/widgets/navigator.js

```diff
@@ -126,6 +126,16 @@ fromPageRect: page coordinates of the origin of the navigation
 */
 NavigatorWidget.prototype.addToHistory = function(title,fromPageRect) {
 	this.wiki.addToHistory(title,fromPageRect,this.historyTitle);
+	this.wiki.setText(this.historyTitle,"focus-tiddler",undefined,title);
```

Comment:
This probably causes an unneeded re-rendering of the story river, since you modify the history list 2 times. Modifying history list needs to be atomic write action

Reply:
Just to clarify that the async nature of the refresh mechanism prevents that from happening: these two writes will just add the `$:/HistoryList` to the list of changes queued up for the next refresh, which actually occurs after an async timeout.